File: SGCN.Web/src/types/certificate.ts
import type { Gender } from './birthRecord';

export type CertificateStatus = 'Active' | 'Annulled';

export interface CertificatePreview {
  childFirstName: string;
  childLastName: string;
  gender: Gender;
  birthDate: string;
  birthTime: string;
  birthPlace: string;
  hospitalName: string;
  communeName: string;
  departmentName: string;
  motherFullName: string;
  fatherFullName?: string | null;
  sgcnId: string;
  requestNumber?: string | null;
}

export interface Certificate {
  id: string;
  certificateNumber: string;
  certificateRequestId?: string | null;
  requestNumber?: string | null;
  birthRecordId: string;
  sgcnId: string;
  childFirstName: string;
  childLastName: string;
  gender: Gender;
  birthDate: string;
  birthTime: string;
  birthPlace: string;
  hospitalName: string;
  communeName: string;
  departmentName: string;
  motherFullName: string;
  fatherFullName?: string | null;
  createdByUserId: string;
  createdByFullName: string;
  status: CertificateStatus;
  verificationCode: string;
  pdfPath?: string | null;
  qrCodePath?: string | null;
  createdAt: string;
  updatedAt?: string | null;
  annulledAt?: string | null;
  annulledReason?: string | null;
}

File: SGCN.Domain/Entities/Certificate.cs
using SGCN.Domain.Common;
using SGCN.Domain.Enums;
using SGCN.Domain.Identity;

namespace SGCN.Domain.Entities;

public sealed class Certificate : BaseEntity
{
    public string CertificateNumber { get; set; } = string.Empty;
    public Guid? CertificateRequestId { get; set; }
    public CertificateRequest? CertificateRequest { get; set; }
    public Guid BirthRecordId { get; set; }
    public BirthRecord BirthRecord { get; set; } = default!;
    public string ChildFirstName { get; set; } = string.Empty;
    public string ChildLastName { get; set; } = string.Empty;
    public Gender Gender { get; set; }
    public DateOnly BirthDate { get; set; }
    public TimeOnly BirthTime { get; set; }
    public string BirthPlace { get; set; } = string.Empty;
    public string HospitalName { get; set; } = string.Empty;
    public string CommuneName { get; set; } = string.Empty;
    public string DepartmentName { get; set; } = string.Empty;
    public string MotherFullName { get; set; } = string.Empty;
    public string? FatherFullName { get; set; }
    public string CreatedByUserId { get; set; } = string.Empty;
    public ApplicationUser CreatedByUser { get; set; } = default!;
    public CertificateStatus Status { get; set; } = CertificateStatus.Active;
    public string VerificationCode { get; set; } = string.Empty;
    public string? PdfPath { get; set; }
    public string? QrCodePath { get; set; }
    public DateTime? AnnulledAt { get; set; }
    public string? AnnulledReason { get; set; }
}
